docs/core/advanced_tutorials/using-result-handlers.md

```diff
@@ -0,0 +1,140 @@
+---
+sidebarDepth: 0
+---
+
+# Using Result Handlers
+
+Let's take a look at using result handlers. As introduced in the concept documentation ["Results and Result Handlers"](../concepts/results.md), Prefect tasks associate their return value with a `Result` object attached to a task's `State`, and they can be extended with `ResultHandler`s that, when enabled, persist results to storage as a pickle into one of the supported storage backends.
+
+Why would you use result handlers in the first place? One common situation is to allow for inspection of intermediary data when debugging flows (including -- maybe especially -- production ones). By checkpointing data during different data transformation Tasks, runtime data bugs can be tracked down by analyzing the checkpoints sequentially.
+
+## Setting up to handle results
+
+The first step to getting the most out of result handlers is to enable results storage. You must enable in two places: globally for the Prefect installation, and at the task level by providing a result handler to tasks either through its flow initialization or as a task-level override. Depending on your storage backend, you may also need to ensure your authentication credentials are set up properly. For Cloud users, some of this is handled by default.
+
+For Core-only users or Cloud users on Prefect versions <0.9.1, you must:
+
+- Opt-in to checkpointing globally by setting the `prefect.config.flows.checkpointing` to "True"
+- Specify the result handler your tasks will use for at least one level of specificity (flow-level or task-level)
+
+For Cloud users on Prefect version 0.9.1+:
+
+- Checkpointing will automatically be turned on; you can disable it by setting `prefect.config.flows.checkpointing` to "False"
```

Comment:
This won't actually turn it off because of this line: https://github.com/PrefectHQ/prefect/blob/master/src/prefect/engine/cloud/task_runner.py#L180

To disable it on a _per task_ basis users can set `checkpoint=False`

Reply:
Updated in https://github.com/PrefectHQ/prefect/pull/2076/commits/6ecd02bc9b0a23947b40ad8249258b781a4a1260